### PR TITLE
issue(#1): Create buildpack for leo

### PR DESCRIPTION
**Pull request by lispking**
issue #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,71 @@
-# leo
+# `gcr.io/amp-buildpacks/leo`
```
Comment (wangeguo) [inline README.md:1]: The domain name is wrong, the correct one is `ghcr.io`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,71 @@
...
+
 A Cloud Native Buildpack for Leo
+
+## Configuration
+
+| Environment Variable      | Description                                                                                                                                                                                                                                                                                       |
+| ------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `$BP_LEO_VERSION` | Configure the version of leo to install. It can be a specific version or a wildcard like `1.*`. It defaults to the latest `1.*` version.                                                                                                                                                  |
```
Comment (wangeguo) [inline README.md:9]: Replace `leo` with `Leo` when it is used as a brand name

File: README.md [hunk 3]
```diff
@@ -1,2 +1,71 @@
...
+| `$BP_LEO_LIBC`    | Configure the libc implementation used by the installed toolchain. Available options: `gnu` or `musl`. Defaults to `gnu` for compatiblity. You do not need to set this option with the Paketo full/base/tiny/static stacks. It can be used for compatibility with more exotic or custom stacks.   |
+| `$BP_ENABLE_LEO_PROCESS`    | Configure the leo launch process, default: `true`. Defaults to `true` for `leo run`.   |
```
Comment (wangeguo) [inline README.md]: Replace `leo` with `Leo` when it is used as a brand name

File: buildpack.toml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    uri = "https://github.com/amp-buildpacks/leo/blob/main/LICENSE"
+
+[metadata]
+  include-files = ["LICENSE", "NOTICE", "README.md", "bin/build", "bin/detect", "bin/main", "buildpack.toml"]
+  pre-package = "scripts/build.sh"
+
+  [[metadata.configurations]]
+    build = true
+    default = "true"
+    description = "enable the leo run process"
```
Comment (wangeguo) [inline buildpack.toml:37]: Replace `leo` with `Leo` when it is used as a brand name

File: buildpack.toml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    name = "BP_ENABLE_LEO_PROCESS"
+
+  [[metadata.configurations]]
+    build = true
+    default = "1.10.0"
+    description = "the leo version"
```
Comment (wangeguo) [inline buildpack.toml:43]: Replace `leo` with `Leo` when it is used as a brand name

File: buildpack.toml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    name = "Leo (musl libc)"
+    purl = "pkg:generic/leo@1.10.0"
+    sha256 = "508264f03760d0a0c9d8cd13c603e0e0d595388b3729762ebfbcc26abe46d667"
+    uri = "https://github.com/AleoHQ/leo/releases/download/v1.10.0/leo-v1.10.0-x86_64-unknown-linux-musl.zip"
+    stacks = ["*"]
+    version = "1.10.0"
+
+    [[metadata.dependencies.licenses]]
+      type = "GNU"
+      uri = "https://github.com/AleoHQ/leo/blob/v1.10.0/LICENSE.md"
```
Comment (wangeguo) [inline buildpack.toml:76]: Will the two dependencies be installed together? Or will it be one or the other based on `$BP_LEO_LIBC`?

File: leo/build.go [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+	cr, err := libpak.NewConfigurationResolver(context.Buildpack, nil)
+	if err != nil {
+		return libcnb.BuildResult{}, fmt.Errorf("unable to create configuration resolver\n%w", err)
+	}
+
+	// create a second time so the configuration is only printed after we know the buildpack should run
+	cr, err = libpak.NewConfigurationResolver(context.Buildpack, &b.Logger)
+	if err != nil {
+		return libcnb.BuildResult{}, fmt.Errorf("unable to create configuration resolver\n%w", err)
+	}
```
Comment (wangeguo) [inline leo/build.go:42]: Is it a duplicate code? (`libpak.NewConfigurationResolver`)

**Review by wangeguo — APPROVED** [review]

**Review by wangeguo — APPROVED** [review]

File: buildpack.toml (hunk 6 above)
Comment (lispking) [reply buildpack.toml:76]: yeah, Leo project have musl and gnu.

File: leo/build.go (hunk 7 above)
Comment (lispking) [reply leo/build.go:42]: rustup code like this.

**Review by wangeguo — APPROVED** [review]

**Pull request merged**